.buildkite/test-check/scripts/image-build-xpu.sh

```
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+set -exo pipefail
+DOCKERFILE=.buildkite/docker/Dockerfile.xpu
+CURRENT_HASH=$(md5sum "$DOCKERFILE" | awk '{print $1}')
+NEED_BUILD=false
+if [[ ! $(docker image ls --format '{{.Repository}}:{{.Tag}}' | grep -Fxq "$IMAGE_NAME") ]]; then
```

Shellcheck (0.11.0)

[style] Use grep -q instead of comparing output with [ -n .. ].

(SC2143)